internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -34,62 +34,223 @@
 package metadata_manager
 
 import (
+	"encoding/json"
+	"path/filepath"
+	"strconv"
+	"time"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
 	"github.com/Azure/azure-storage-fuse/v2/internal/dcache"
 )
 
 // BlobMetadataManager is the implementation of MetadataManager interface
 type BlobMetadataManager struct {
-	cacheDir string
+	cacheDir         string
+	storageCallbacks dcache.StorageCallbacks
 }
 
 // NewMetadataManager creates a new implementation of the MetadataManager interface
-func NewMetadataManager(cacheDir string) (MetadataManager, error) {
+func NewMetadataManager(cacheDir string) (*BlobMetadataManager, error) {
```

Comment:
as mentioned in the chat, don't export this function, instead call it from the init() function in this file, so that users can use it as 

import mm "path/to/metadata/manager/package"

mm.CreateFileInit()